rfcs/041-item-statuses/README.md

```diff
@@ -0,0 +1,311 @@
+# RFC 041: Statuses in the items API
+
+As part of the [requesting API](../039-requesting-api-design), we need to present an item status.
+We already have a model that we'll use:
+
+```json
+"status": {
+  "id" : "available",
+  "label" : "Available",
+  "type": "ItemStatus"
+}
+```
+
+but we need to decide what value it can take.
+
+This RFC proposes a set of values that we will present through this API.
+
+
+
+## Guiding principles
+
+1.  **We should not assume requests are sent to a particular service.**
+
+    Although the initial implementation will be making requests for items in Sierra, we may add support for requesting from other systems in a future update.
+    We should not assume the use of Sierra, or of Sierra item statuses.
+
+2.  **A user's list of holds is private.**
+
+    Alice should not be able to find out what items Bob has placed on hold, and vice versa.
+
+3.  **The status information will be publicly visible, and displayed on the works page.**
+
+    For consistency, we should use the same terminology as will be used on the works page.
+    We should assume users will see these statuses, and not expose information about the library workings.
+
+
+
+## Existing Sierra statuses
+
+This section describes the existing status data in Sierra items.
+
+Sierra presents the item status in the [Item STATUS field](https://documentation.iii.com/sierrahelp/Default.htm#sril/sril_records_fixed_field_types_item.html#STATUS).
+There are standard codes supplied as part of Sierra, but libraries can define additional item status values.
+
+This is a list of item statuses in use at Wellcome, based on the reporting cluster data.
+(Deleted and suppressed items have been )
+
+<table>
+  <tr>
+    <th>code</th>
+    <th>name</th>
+    <th>count</th>
+    <th>comment</th>
+  </tr>
+  <tr>
+    <td><code>-</code></td>
+    <td>Available</td>
+    <td>616,145</td>
+    <td>
+      The Sierra docs say this status is "items that can circulate", and that "the item retains this status even when checked out".
+      We should test with this use case, and make sure the items API responds appropriately.
+    </td>
+  </tr>
+  <tr>
+    <td><code>b</code> / <code>c</code></td>
+    <td>As above</td>
```

Comment:
Optimism: can we infer what "above" was/is?

Reply:
Ish.

We can't get the item ordering from the Sierra API, so we can't tell what the "above" item is.

We do have a workaround in the catalogue pipeline, because you get similar "bound in above" or "contained in above" in the location field. We look at all the other items for locations that aren't "X in above", and if there's only one distinct location among them, we assume the "X in above" refers to that, and use that instead.

See https://github.com/wellcomecollection/catalogue-pipeline/blob/master/pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/SierraItems.scala#L55-L90

That's not perfect because:

* Some bibs only have a single item that's "as above", so there are no other items we can look at
* I would be a bit more wary of this for status, where items can presumably move out of sync with each other

I've heard vague mutterings from C&I that they might consider ditching this, and this might be another reason to do so.

Reply:
In the meantime do we remove this status and copy the status from the [as best can be determined] "above" item?

Reply:
Hmm. This is what that looks like in the OPAC:

<img width="1067" alt="Screenshot 2021-05-07 at 14 56 40" src="https://user-images.githubusercontent.com/301220/117460494-7c1f6000-af44-11eb-8274-667bd38e65c0.png">

Notice you aren't allowed to request an item that's "bound in above" – this is set by the "Rules for requesting" file.

Because we can't order these sensible, I'm not sure what's best to do here. Possibly these get folded into "not-requestable"?

Reply:
The problem is they are requestable, they're just requestable by requesting another item.